mutex_protected_test.cc

```diff
@@ -172,14 +172,58 @@ TYPED_TEST(MutexProtectedTest, LockMultiple) {
   mutex_protected<int, TypeParam> a(1);
   mutex_protected<int, TypeParam> b(2);
   {
-    auto [la, lb] = xyz::lock(a, b);
+    auto [la, lb] = xyz::lock_protected(a, b);
     EXPECT_EQ(*la, 1);
     EXPECT_EQ(*lb, 2);
     *la += 10;
     *lb += 10;
   }
   {
-    auto [lb, la] = xyz::lock(b, a);
+    auto [lb, la] = xyz::lock_protected(b, a);
+    EXPECT_EQ(*la, 11);
+    EXPECT_EQ(*lb, 12);
+  }
+}
+
+TYPED_TEST(MutexProtectedTest, TryLockMultiple) {
+  mutex_protected<int, TypeParam> a(1);
+  mutex_protected<int, TypeParam> b(2);
+  {
+    auto r = xyz::try_lock_protected(a, b);
+    ASSERT_TRUE(r.has_value());
+
+    auto& [la, lb] = r.value();
+    EXPECT_EQ(*la, 1);
+    EXPECT_EQ(*lb, 2);
+    *la += 10;
+    *lb += 10;
+  }
+  if constexpr (!std::is_same_v<TypeParam, std::recursive_mutex> &&
```

Comment:
Hmm, maybe it'd be better to add a NonRecursiveMutexProtectedTest with a different subset of mutexes.

Reply:
As you prefer. Problem fixed at any rate.